Tribler/community/tunnel/tunnel_community.py

```diff
@@ -758,6 +799,28 @@ def _ours_on_created_extended(self, circuit, message):
             from Tribler.Core.simpledefs import NTFY_TUNNEL, NTFY_CREATED, NTFY_EXTENDED
             self.notifier.notify(NTFY_TUNNEL, NTFY_CREATED if len(circuit.hops) == 1 else NTFY_EXTENDED, circuit)
 
+
+    def on_introduction_request(self, messages):
+        exitnode = self.settings.become_exitnode
+        extra_payload = [exitnode]
+        super(TunnelCommunity, self).on_introduction_request(messages, extra_payload)
+        for message in messages:
+            pubkey = message.candidate.get_member().public_key
+            connectable = message.candidate.connection_type == u"public"
+            self.exit_candidates[pubkey] = message.payload.exitnode and connectable
```

Comment:
Hi Niels,
What do you exactly mean with removing the doctor, just yielding verified
peers? The exitnode flag for each peer that gets introduced is the most
important thing to record here, and connectability is an extra constraint
on this, should connectability be checked somewhere else?

Op donderdag 26 maart 2015 heeft Niels Zeilemaker <notifications@github.com>
het volgende geschreven:

> In Tribler/community/tunnel/tunnel_community.py
> <https://github.com/Tribler/tribler/pull/1325#discussion_r27189192>:
>
> > @@ -758,6 +799,28 @@ def _ours_on_created_extended(self, circuit, message):
> >              from Tribler.Core.simpledefs import NTFY_TUNNEL, NTFY_CREATED, NTFY_EXTENDED
> >              self.notifier.notify(NTFY_TUNNEL, NTFY_CREATED if len(circuit.hops) == 1 else NTFY_EXTENDED, circuit)
> >
> > +
> > +    def on_introduction_request(self, messages):
> > +        exitnode = self.settings.become_exitnode
> > +        extra_payload = [exitnode]
> > +        super(TunnelCommunity, self).on_introduction_request(messages, extra_payload)
> > +        for message in messages:
> > +            pubkey = message.candidate.get_member().public_key
> > +            connectable = message.candidate.connection_type == u"public"
> > +            self.exit_candidates[pubkey] = message.payload.exitnode and connectable
>
> Then just remove the doctor and simply switch to yield verified peers,
> those have the connection type flag
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Tribler/tribler/pull/1325/files#r27189192>.
>